modules/GKB/Release/Step.pm

```diff
@@ -197,474 +199,494 @@ use GKB::Release::Config;
+        open my $pre_step_test_fh, '>', $pre_step_test_log;
+        print {$pre_step_test_fh} (join "\n", @pre_step_test_errors);
+        close $pre_step_test_fh;
+        say releaselog("ERRORS from $self->{name} pre-step tests reported -- see $pre_step_test_log");
+
+        return; # Prevent step from being run by stopping the run method prematurely
     } else {
-		say releaselog("No errors from $self->{name} pre-step tests");
-	}
-	
-	$self->run_commands($self->gkb);
-	
-	say "Running $self->{name} post-step tests...";
-	my @post_step_test_errors = $self->post_step_tests();
-	if (@post_step_test_errors) {
-		
-		# Let's write the errors to a file. That way, someone OTHER than the mail recipient can see them.
-		my $post_step_test_log = File::Spec->catfile($self->directory, 'post_step_test_errors.log');
-		open(my $post_step_test_fh, '>', $post_step_test_log);
-		binmode $post_step_test_fh, ":utf8";
-		print $post_step_test_fh join("\n", @post_step_test_errors);
-		close $post_step_test_fh;
-		say releaselog("ERRORS from $self->{name} post-step tests reported -- see $post_step_test_log");
-		
-		say "Errors from $self->{name} post-step tests -- sending e-mail";
-		$self->mail->{'body'} = "Errors Reported\n\n" . join("\n", @post_step_test_errors);
-		$self->mail->{'to'} = 'automation';
-	} else {
-		say "No errors from $self->{name} post-step tests";
-		$self->mail->{'body'} .= "\n\n" if $self->mail->{'body'};
-		$self->mail->{'body'} .= "$self->{name} step has completed successfully";
-	}
-	$self->mail_now();
-	
-	say releaselog("Archiving output, logs, dump files...");
-	$self->archive_files($version);
-
+        say releaselog("No errors from $self->{name} pre-step tests");
+    }
+
+    $self->run_commands($self->gkb);
+
+    say "Running $self->{name} post-step tests...";
+    my @post_step_test_errors = $self->post_step_tests();
+    if (@post_step_test_errors) {
+        # Let's write the errors to a file. That way, someone OTHER than the mail recipient can see them.
+        my $post_step_test_log = File::Spec->catfile($self->directory, 'post_step_test_errors.log');
+        open my $post_step_test_fh, '>', $post_step_test_log;
+        binmode $post_step_test_fh, ':encoding(UTF-8)';
+        print {$post_step_test_fh} join "\n", @post_step_test_errors;
+        close $post_step_test_fh;
+        say releaselog("ERRORS from $self->{name} post-step tests reported -- see $post_step_test_log");
+
+        say releaselog("Errors from $self->{name} post-step tests -- sending e-mail");
+        $self->mail->{'body'} = "Errors Reported\n\n" . join("\n", @post_step_test_errors);
+        $self->mail->{'to'} = 'automation';
+    } else {
+        say releaselog("No errors from $self->{name} post-step tests");
+        if ($self->mail->{'body'}) {
+            $self->mail->{'body'} .= "\n\n";
+        }
+        $self->mail->{'body'} .= "$self->{name} step has completed successfully";
+    }
+    $self->mail_now();
```

Comment:
Is there any way to catch errors if mailing fails?

Reply:
Yup, I'm sure there's a way to do that.  I'll look into it.

Reply:
Okay, I've added in some logic to capture mail errors and report them to the release log and on screen while running.

Reply:
:+1: